Library/Tracking/KSRAnalytics.swift

```diff
@@ -379,6 +379,26 @@ public final class KSRAnalytics {
     case watch
     case watched
 
+    /**
+     Initialize `TypeContext` with `DiscoveryParams`
+     to determine type context for discovery collection filters.
+
+     - parameter params: a `DiscoveryParams` object
+     */
+    init?(params: DiscoveryParams) {
```

Comment:
After making the changes below, you can remove the `?` here. It will no longer need to be a failable init.